arbnode/transaction_streamer.go

```diff
@@ -544,6 +544,18 @@ func (s *TransactionStreamer) GetMessage(seqNum arbutil.MessageIndex) (*arbostyp
 		return nil, err
 	}
 
+	err = message.Message.FillInBatchGasCost(func(batchNum uint64) ([]byte, error) {
```

Comment:
We want to make absolutley sure that in every way we can get a message to consensus client, it will always have the batchGasCost Filled.

I think there are other cases we want to cover, specifically:
legacyGetDelayedMessageAndAccumulator
GetDelayedMessageAccumulatorAndParentChainBlockNumber

Reply:
Covered those other cases.

We don't have tests covering L1MessageType_BatchPostingReport today, right?